### PR TITLE
script for puppeteer rasterizing

### PR DESCRIPTION
**Pull request by benny-conn**
Changes:

- **Added** script for rasterizing svgs into images or gifs

TODO:

- [x] Testing different types of svgs

### CODE REVIEW

File: scripts/rasterize_svg.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      await page.setViewport({ width: viewport.width, height: viewport.height });
+    }
+  });
+
+  await page.goto(url);
+
+  const frames = [];
+  for (let i = 0; i < totalFrames; i++) {
+    const frame = await captureFrame(page, 100); // Capture frame every 100ms
+    frames.push(PNG.sync.read(frame));
```
Comment (tbezman) [inline scripts/rasterize_svg.js:34]: @benny-conn are the gifs going to look super choppy with 100ms waits? what are the tradeoffs here?

**Review by radazen — APPROVED** [review]
This is so cool, Benny! Awesome solution to the SVG problem. I left a few comments!

One other thought: have you tried any fast-moving animations with this? It looks to me like the timing of captures is:

```
  every 30 milliseconds
+ time taken to capture screenshot
+ any variations in the waitForTimeout timing*
```

\* -  there might be variation here since delay functions usually guarantee something like "this will wait for AT LEAST 30 milliseconds, but might wait a little longer before returning control to you"

I'm wondering if capturing frames at "roughly every 30ms" and animating them in a GIF at "exactly every 30ms" will end up feeling jerky or unsettling.

File: scripts/rasterize_svg.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    let gifBuffer = Buffer.alloc(0);
+    stream.on('data', (chunk) => (gifBuffer = Buffer.concat([gifBuffer, chunk])));
+    stream.on('end', () => {
+      console.log('GIF');
+      console.log(gifBuffer.toString('base64'));
+    });
+
+    encoder.start();
+    encoder.setRepeat(0); // 0 for repeat, -1 for no-repeat
+    encoder.setDelay(500); // frame delay in ms
```
[the remark this answers is not in the document]
Comment (benny-conn) [reply scripts/rasterize_svg.js:74]: Good catch!

File: scripts/rasterize_svg.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      let viewport = svg.getBoundingClientRect();
+      await page.setViewport({ width: viewport.width, height: viewport.height });
+    }
+  });
+
+  await page.goto(url);
+
+  const frames = [];
+  for (let i = 0; i < totalFrames; i++) {
+    const frame = await captureFrame(page, 30); // Capture frame every 100ms
```
[the remark this answers is not in the document]
Comment (benny-conn) [reply scripts/rasterize_svg.js:33]: Yep!

**Comment by jarrel-b** [comment]
Would this be a more reliable way of getting an SVG's dimensions than our current method? I noticed in the pipeline that svg dimensions are occasionally mssing, would it be worth it to use this method going forward?

**Comment by benny-conn** [comment]
> Would this be a more reliable way of getting an SVG's dimensions than our current method? I noticed in the pipeline that svg dimensions are occasionally mssing, would it be worth it to use this method going forward or as a fallback to the current method?

Yes it would be a better way! Much easier to get dimensions of GIF and PNG.

**Pull request merged**